homeworks/homework2/task3/smart_args.py

```diff
@@ -0,0 +1,46 @@
+from copy import deepcopy
+from typing import TypeVar, Generic, Optional, Callable
+
+T = TypeVar("T", covariant=True)
```

Comment:
Why you define T as a covariant type? As I see, you define some functions where T used as a consumer therefore it can't be covariant.
Btw, pep recommends using the `_co` prefix for covariant types, e.g. `T_co`.

Reply:
You're right, I don't need to define T as a covariant type here. Initially, I didn't quite understand its purpose correctly.